analyses/tmb-vaf-longitudinal/01-preprocess-data.Rmd

```diff
@@ -0,0 +1,201 @@
+         timepoints_deceased = match_id_bs,
+         timepoints_deceased = case_when(grepl("Diagnosis", match_id_bs) ~ "1", 
+                                         grepl("Recurrence", match_id_bs) ~ "2",
+                                         grepl("Progressive", match_id_bs) ~ "3",
+                                         TRUE ~ timepoints_deceased),
+         timepoints_other = match_id_bs,
+         timepoints_other = case_when(grepl("Deceased", match_id_bs) ~ "4",
+                                      TRUE ~ timepoints_other)) %>%
+  unique() %>%
+  write_tsv(file.path(scratch_dir, "maf_autopsy.tsv"))
+
+# How many patients have paired time samples with autopsy time point present?
+paired_samples_autopsy <- length(unique(maf_autopsy$Kids_First_Participant_ID))
+```
+
+## How many patient cases with autopsy samples in the paired longitudinal data?
+There are `r paired_samples_autopsy` autopsy samples out of the total `r paired_samples` patient cases.
+
+# Create TMB file for all patient samples with genomic assays
+```{r tmb-genomic}
+# We will create similar unique identifiers as we did for the maf file.
+# We will also create identifiers for each timepoint (timepoints_deceased and timepoints_other).
+tmb_genomic <- patient_list_metadata %>% 
+  left_join(tmb_all, by = c("Kids_First_Biospecimen_ID", "experimental_strategy")) %>%
+  filter(!is.na(Kids_First_Participant_ID),
+         !is.na(tmb)) %>%
+   mutate(match_id_bs = paste(tumor_descriptor, Kids_First_Biospecimen_ID, sep = "_"),
+          timepoints_deceased = match_id_bs,
+          timepoints_deceased = case_when(grepl("Diagnosis", match_id_bs) ~ "1",
+                                          grepl("Recurrence", match_id_bs) ~ "2",
+                                          grepl("Progressive", match_id_bs) ~ "3",
+                                          TRUE ~ timepoints_deceased),
+          timepoints_other = match_id_bs,
+          timepoints_other = case_when(grepl("Deceased", match_id_bs) ~ "4",
+                                       TRUE ~ timepoints_other)) 
+
+tmb_paired_samples <- length(unique(tmb_genomic$Kids_First_Participant_ID))
+
+# There are missing tmb information for some `Kids_First_Biospecimen_ID` in the pbta cohort.
+# This will result into cases with no matched longitudinal samples (`tumor_descriptor`) per `Kids_First_Participant_ID`.
+# First, let's find out which `Kids_First_Biospecimen_ID` samples are missing from the tmb table.
+bs_missing_tmb <- pbta %>% 
+  left_join(tmb_all, by = c("Kids_First_Biospecimen_ID", "experimental_strategy")) %>% 
+  filter(is.na(tmb)) %>%
+  select(Kids_First_Biospecimen_ID) %>% 
+  write_tsv(file.path(results_dir, "bs_missing_tmb.tsv"))
+
+# How many biospecimen samples are missing from the TMB file?
+bs_missing_tmb_samples <- length(unique(bs_missing_tmb$Kids_First_Biospecimen_ID))
+
+# How many of those are missing from the VAF file?
+bs_missing_tmb_vaf <- maf %>% 
+  select(Kids_First_Biospecimen_ID) %>%
```

Comment:
I think filtering base don `VAF` needs to be performed here...right now this is just resulting in the same samples as are in `bs_missing_tmb`

Reply:
In this code, we were checking if the bs_id samples missing from the TMB file were also missing from the VAF file or if there were any more samples. They were the same in this case hence the files matching. This is also explained in the `Notes` section `Lines 194-197`.

Reply:
Also, see this in lines 174-175:

```
# Are the same biospecimen samples missing in both df?
print(diff <- setdiff(bs_missing_tmb_vaf, bs_missing_tmb))
```